vxquery-core/src/main/java/org/apache/vxquery/xmlquery/translator/XMLQueryTranslator.java

```diff
@@ -1484,11 +1499,14 @@ private LogicalVariable translatePathExpr(PathExprNode pe, TranslationContext tC
                     throw new IllegalStateException("Unknown path node: " + pathNode.getTag());
                 }
                 if (predicates != null && !predicates.isEmpty()) {
+                    ctxExpr = vre(createAssignment(ctxExpr, tCtx));
                     ctxExpr = sfce(asc ? BuiltinOperators.SORT_DISTINCT_NODES_ASC_OR_ATOMICS
                             : BuiltinOperators.SORT_DISTINCT_NODES_DESC_OR_ATOMICS, ctxExpr);
                     for (ASTNode pn : predicates) {
-                        tCtx = tCtx.pushContext();
-                        tCtx.pushVariableScope();
+                        if (!popScope) {
```

Comment:
This if statement should be removed. I will have code that actually makes this work in the near future.